File: Src/Icm.Core.Tests/Basic types extensions/NullableExtensionsTest.vb
<TestFixture()>
Public Class NullableExtensionsTest

    <Test()>
    Public Sub VTest()
        Dim obj As Integer?

        obj = 3
        Assert.AreEqual(obj.V, obj.Value)

        obj = Nothing
        Assert.Throws(Of InvalidOperationException)(
            Sub()
                Dim a = obj.V
            End Sub)
    End Sub


    <Test()>
    Public Sub HasNotValueTest()
        Dim obj As Integer?

        obj = 3
        Assert.AreEqual(obj.HasNotValue, Not obj.HasValue)

        obj = Nothing
        Assert.AreEqual(obj.HasNotValue, Not obj.HasValue)
    End Sub

End Class

File: Src/Icm.Core.Tests/Basic types extensions/NullableExtensionsTest.cs
using System;
using Icm;
using NUnit.Framework;

[TestFixture()]
public class NullableExtensionsTest
{

	[TestCase(3)]
	public void V_Test(int? obj)
	{
		Assert.That(obj.V(), Is.EqualTo(obj.Value));
	}
    
    [TestCase(null)]
    public void V_Test2(int? obj)
    {
        Assert.That(() => obj.V(), Throws.InvalidOperationException);
    }

    [TestCase(3)]
	[TestCase(null)]
	public void HasNotValue_Test(int? obj)
	{
		Assert.That(obj.HasNotValue(), Is.EqualTo(!obj.HasValue));
	}

}
